From cc917e074cd23a1630cb0eb4b7f3ec6ba2777d80 Mon Sep 17 00:00:00 2001
From: Stefan Vigerske <svigerske@gams.com>
Date: Mon, 16 Feb 2026 05:00:52 +0100
Subject: [PATCH] prefix also once_flag and call_once with TNY

- to avoid conflict with pthread of modern glibc 2.43
- closes scipopt/scip#191
---
 src/tinycthread/tinycthread.c |  2 +-
 src/tinycthread/tinycthread.h | 12 ++++++------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/tinycthread/tinycthread.c b/src/tinycthread/tinycthread.c
index 722f89a495..75fc42f985 100644
--- a/src/tinycthread/tinycthread.c
+++ b/src/tinycthread/tinycthread.c
@@ -895,7 +895,7 @@ int _tthread_timespec_get(struct timespec *ts, int base)
 #endif /* _TTHREAD_EMULATE_TIMESPEC_GET_ */
 
 #if defined(_TTHREAD_WIN32_)
-void call_once(once_flag *flag, void (*func)(void))
+void TNY_call_once(TNY_once_flag *flag, void (*func)(void))
 {
   /* The idea here is that we use a spin lock (via the
      InterlockedCompareExchange function) to restrict access to the
diff --git a/src/tinycthread/tinycthread.h b/src/tinycthread/tinycthread.h
index 365c2e2b58..fdc34ea98b 100644
--- a/src/tinycthread/tinycthread.h
+++ b/src/tinycthread/tinycthread.h
@@ -447,11 +447,11 @@ int tss_set(tss_t key, void *val);
   typedef struct {
     LONG volatile status;
     CRITICAL_SECTION lock;
-  } once_flag;
-  #define ONCE_FLAG_INIT {0,}
+  } TNY_once_flag;
+  #define TNY_ONCE_FLAG_INIT {0,}
 #else
-  #define once_flag pthread_once_t
-  #define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
+  #define TNY_once_flag pthread_once_t
+  #define TNY_ONCE_FLAG_INIT PTHREAD_ONCE_INIT
 #endif
 
 /** Invoke a callback exactly once
@@ -460,9 +460,9 @@ int tss_set(tss_t key, void *val);
  * @param func Callback to invoke.
  */
 #if defined(_TTHREAD_WIN32_)
-  void call_once(once_flag *flag, void (*func)(void));
+  void TNY_call_once(TNY_once_flag *flag, void (*func)(void));
 #else
-  #define call_once(flag,func) pthread_once(flag,func)
+  #define TNY_call_once(flag,func) pthread_once(flag,func)
 #endif
 
 #ifdef __cplusplus
